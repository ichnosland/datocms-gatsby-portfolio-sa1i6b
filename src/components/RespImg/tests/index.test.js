import React from 'react';
import { shallow } from 'enzyme';

import { RespImg, DesktopImage, MobileImage } from '../index';

describe('<RespImg />', () => {
  it('should render a <img> tag', () => {
    const renderedComponent = shallow(<RespImg />);
    expect(renderedComponent.type()).toEqual('img');
  });

  it('should render its css w/ props maxWidth', () => {
    const maxWidth = '30%';
    const renderedComponent = shallow(<RespImg maxWidth={maxWidth} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render css', () => {
    const renderedComponent = shallow(<RespImg maxWidth="60%" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<DesktopImage />', () => {
  it('should render a RespImg', () => {
    const renderedComponent = shallow(<DesktopImage />);
    expect(renderedComponent.find(RespImg).length).toEqual(1);
  });

  it('should render its css w/ props border', () => {
    const border = '1px solid green';
    const renderedComponent = shallow(<RespImg border={border} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props margin', () => {
    const margin = 'auto';
    const renderedComponent = shallow(<RespImg margin={margin} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render css', () => {
    const renderedComponent = shallow(<DesktopImage margin="20px" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render css', () => {
    const renderedComponent = shallow(<DesktopImage border="2px dotted green" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<MobileImage />', () => {
  it('should render a RespImg', () => {
    const renderedComponent = shallow(<MobileImage />);
    expect(renderedComponent.find(RespImg).length).toEqual(1);
  });

  it('should render its css w/ props border', () => {
    const border = '1px solid green';
    const renderedComponent = shallow(<RespImg border={border} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props margin', () => {
    const margin = 'auto';
    const renderedComponent = shallow(<RespImg margin={margin} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render css', () => {
    const renderedComponent = shallow(<MobileImage margin="20px" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render css', () => {
    const renderedComponent = shallow(<MobileImage border="2px dotted green" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
