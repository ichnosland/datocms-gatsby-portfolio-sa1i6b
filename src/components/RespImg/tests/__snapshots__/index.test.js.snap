// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopImage /> should render css 1`] = `
.c0 {
  display: block;
  margin: 20px;
  border: 3px solid #D9D9D9;
}

@media (max-width:48em) {
  .c0 {
    display: none;
  }
}

<RespImg
  className="c0"
  margin="20px"
/>
`;

exports[`<DesktopImage /> should render css 2`] = `
.c0 {
  display: block;
  margin: 30px auto;
  border: 2px dotted green;
}

@media (max-width:48em) {
  .c0 {
    display: none;
  }
}

<RespImg
  border="2px dotted green"
  className="c0"
/>
`;

exports[`<DesktopImage /> should render its css w/ props border 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
}

<img
  className="c0"
/>
`;

exports[`<DesktopImage /> should render its css w/ props margin 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
}

<img
  className="c0"
/>
`;

exports[`<MobileImage /> should render css 1`] = `
.c0 {
  display: block;
  margin: 20px;
  border: 3px solid #D9D9D9;
}

@media (min-width:48.0625em) {
  .c0 {
    display: none;
  }
}

<RespImg
  className="c0"
  margin="20px"
/>
`;

exports[`<MobileImage /> should render css 2`] = `
.c0 {
  display: block;
  margin: 30px auto;
  border: 2px dotted green;
}

@media (min-width:48.0625em) {
  .c0 {
    display: none;
  }
}

<RespImg
  border="2px dotted green"
  className="c0"
/>
`;

exports[`<MobileImage /> should render its css w/ props border 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
}

<img
  className="c0"
/>
`;

exports[`<MobileImage /> should render its css w/ props margin 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
}

<img
  className="c0"
/>
`;

exports[`<RespImg /> should render css 1`] = `
.c0 {
  width: 100%;
  max-width: 60%;
}

<img
  className="c0"
/>
`;

exports[`<RespImg /> should render its css w/ props maxWidth 1`] = `
.c0 {
  width: 100%;
  max-width: 30%;
}

<img
  className="c0"
/>
`;
